docs/v1/P4Runtime-Spec.mdk

```diff
@@ -6598,4 +6680,53 @@ clients are aware that the server is using a larger maximum receive message
 size. The gRPC server running the P4Runtime service must not set the maximum
 receive message size to a value smaller than the default (4MB).
 
+## P4Runtime Entries files { #sec-entries-files }
+
+The open source P4 compiler `p4c` [@p4c] implements an option to
+generate an "entries file", &ie; a file that contains all table
+entries declared via the `entries` table property within the program.
+
+An example P4~16~ program that can be used to demonstrate this
+capability is `table-entries-ternary-bmv2.p4` [@p4cTestProgramForConstEntries]:
+
+    git clone https://github.com/p4lang/p4c
+    cd p4c/testdata/p4_16_samples
+    mkdir tmp
+    p4test --arch v1model \
+        --p4runtime-files tmp/p4info.txt \
+        --p4runtime-entries-files tmp/entries.txt \
+        table-entries-ternary-bmv2.p4
+
+You can replace the `.txt` suffix of the file name `tmp/entries.txt`
+in the example command above with `.json` or `.bin`.  The `.bin`
+format is a binary P4Runtime API protobuf message format.  The `.txt`
+format is the text encoding of the same Protobuf messages.
+
+Target devices are *not* required to use this file.  For example, if a
+target has a P4 compiler back end that encodes all of the necessary
+details from the P4 source program, including the `entries` of tables,
+in a target-specific binary format, then that target might have no
+reason to generate these entries files.
+
+Some target devices might choose to generate entries files, and also
+to require doing so in order to have a correct implementation.  For
+example, a target runtime implementation might take a target-specific
+binary format for the compiled P4 program that does *not* contain any
+data describing the `entries` of tables, plus the entries file
+generated by `p4c`, and use the entries file to load the initial
+entries of tables into the device.
+
+The format of the entries file is a single `WriteRequest` message
+containing one `Update` sub-message per entry in the P4 source program
+defined via an `entries` table property.  All `Update` sub-messages
+have `type` equal to `INSERT`, and `entity` is a `TableEntry` message
+containing the data for one table entry.
+
+Note that for a table with `const entries`, or with `entries` having
+one or more individual entries declared `const`, it would be an error
+for a P4Runtime client to send a `WriteRequest` to a P4Runtime server
+with the contents of the entries file, because the server should
```

Comment:
It may be good to further clarify that error applies to the entries with const only, not the whole contents of the entries file.

Reply:
With the latest commit on this PR as of writing this comment, I have replaced the last paragraph of the new appendix with the following, in an attempt to clarify:
```
Note that if a P4Runtime client attempted to send a `WriteRequest` to
a P4Runtime server with the contents of the entries file, the server
must return an error for each entry that has `is_const` true, as
described in Section [#sec-table-entry].
```